### DISCUSSION TITLE
GPTH stuck at Finding Duplicates

### DISCUSSION THREAD

**Discussion by sghulyani** [TheLastGimbus/GooglePhotosTakeoutHelper#263] [I NEED HELP]
Hi. I have about 350Gb (495,283 files in 981 Folders. Total 9 folders - around 50Gb each) in my Takeout folder . After running GPTH, it is stuck on Finding Duplicates for about 4 hours. Is it normal for it to take that much time? I have AMD Ryzen 7000 series with 16Gb Ram. 
Also tried on my Macbook Pro 2017. Its been 2 hours and still stuck on Finding Duplicates. Can you please guide how much time does it normally take to pass this step? Or is it actually stuck?
Am doing the conversion on my external HDD.

**Comment by TheLastGimbus** [comment]
Oh may... that's an interesting case

Finding duplicates is pretty unperdicatble - first, gpth groups files by their size, and then calculates hashesh between them... and *then* removes one if many have same hash - i cannot realy perdict how many hashes will be calculated, or how much has left, so I left it like that...

Your setup seems pretty fast, (tho if it's an actual magnetical HDD, then this is the thing that bottlenecks everthing) - but also your takeout seems pretty damn big 😳  - could you try to leave it for few more hours 😅😅 ? If it still won't work, i will make a temporary version of gpth to test it with you 👀

**Reply by sghulyani** [reply]
Thank you for the response. It did not work even after keeping it live for a couple of more hours. I finally did it directly with exiftool.

**Reply by TheLastGimbus** [reply]
Happy that you resolved your issue, but if you could try to find some more debug info I may resolve the issue for others (feel free to leave this if you don't have time tho)

Could you try to look at ram/cpu usage etc while this is ongoing? To see if there's a moment when it quietly crashes or something?
